### ISSUE TITLE
Twitter \ OG \ Header Images [Feature Request] 

### ISSUE THREAD

**Issue by infominer33** [AngeloStavrow/indigo#44]
First of all, let me say THANK YOU, for this Gorgeous Indieweb enabled Hugo Theme! I came over from Jekyll just to get here :)

* https://infominer.id/web-work/
* https://github.com/infominer33/web-work

**Is your feature request related to a problem? Please describe.**

Twitter \ OG Cards are not properly populating.

I also would love the option to include a feature image both on the homepage and in the post.

**Describe the solution you'd like**

* https://developer.twitter.com/en/docs/tweets/optimize-with-cards/guides/getting-started

I'm new to hugo, so there's likely a popular plugin or include I can insert, but I haven't found one to work, yet.

**Describe alternatives you've considered**
I'm open to anything, I'll be exploring the question more deeply, but wanted to say thank you as much as ask for a feature request.

**Additional context**

* https://infominer.id/web-work/tags/indieweb/

Thanks again! Look forward to getting to know your theme, and indieweb better!

**Comment by AngeloStavrow** [comment]
Thanks for opening this! You're right that generally speaking, meta tags for the site aren't populating properly — I recently discovered https://www.heymeta.com/ as a good way to verify this working on another site, and there are a bunch of things that could be improved.

**Comment by infominer33** [comment]
yes, I've been discovering that there is code for that, but haven't examined it closely.

I'm pretty new to static site development, so I'd eventually dig in to see what could be done with it. Just lately I've been seeing what indieweb themes are available for whatever SSG, and seeing where they're at collectively... and let my interest in indieweb drive my heaviest habit: independent web-publishing.

I'm most excited with learning how to make use of all the awesome indieweb markup you ensure each page is wrapped with :D

**Comment by infominer33** [comment]
So, would probably need some tweaking for hugo... 

However, I just dropped this include into 
[Jekyll-Indieweb](https://github.com/miklb/jekyll-indieweb) and it works great!

* [Minimal Mistakes SEO include](https://github.com/mmistakes/minimal-mistakes/blob/master/_includes/seo.html)
* [Associated (not extremely thorough) documentation](https://mmistakes.github.io/minimal-mistakes/docs/configuration/#twitter-cards-and-facebook-open-graph)

**Comment by AngeloStavrow** [comment]
Ah that seems pretty straightforward to port, nice find!

**Comment by infominer33** [comment]
now we're cooking w grease!

https://github.com/mikeymop/minimal-mistakes-hugo/blob/master/hugo/minMistakes/themes/minMistakes/layouts/partials/seo.html

I figured out the other day there is a hugo version of minimal mistakes, the creator built for both platforms at once..

anyways, his is the gold standard as far as Jekyll themes go... pretty sure all that's required is to keep a copy of his original license to use for any purpose, tho I'm not 100% on the particulars, cause I've never shipped anything where I needed to know.

**Comment by infominer33** [comment]
Ok, **Now** I have figured out twitter cards which choose the image from the config file (if params:images is set) and\or the page level if `images` variable provided in the front matter:

```
    <!-- Twitter Cards-->
    <!-- Twitter summary card with large image must be at least 280x150px -->
    <meta name="twitter:card" content="summary_large_image"/>
    <meta name="twitter:image:src" content="{{ with .Params.images }}{{ . }}{{ else }}{{ .Site.Params.images }}{{ end }}"/>
    <!-- Twitter Card data -->
    <meta name="twitter:title" content="{{ .Title }}"/>
    <meta name="twitter:description" content="{{ with .Description }}{{ . }}{{ else }}{{if .IsPage}}{{ .Summary }}{{ else }}{{ with .Site.Params.description }}{{ . }}{{ end }}{{ end }}{{ end }}"/>
    {{ with .Site.Social.twitter }}<meta name="twitter:site" content="@{{ . }}"/>{{ end }}
    {{ with .Site.Social.twitter_domain }}<meta name="twitter:domain" content="{{ . }}"/>{{ end }}
```

this is now live on my site:

https://github.com/web-work-tools/indieweb/commit/1cc418a0e278a87bccb88799c8e0bc11c8405124

which can be verified by checking the source of https://web-work.tools/indieweb and that of https://web-work.tools/indieweb/resources

I didn't include in a pull request, since this example assumes you have title description and image set in the params of config.yml

It's simple and gets the job done

**Comment by infominer33** [comment]
I just wanted to include, because it's not obvious if you don't visit the site, but I didn't just pass that image variable through to the og_image, but also have set a header image on each page.

I don't know enough CSS to make this proper, but I'm happy with it for now.. I feel like I got quite a lot of value from adding a few simple lines of code.

## Example on a Post
![](https://imgur.com/BFI8c78.png)

## On the Home Page
![](https://imgur.com/XcQt2fB.png)
